src/LetsEncrypt.Azure/AzureKeyVaultExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Nate McMaster.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using McMaster.AspNetCore.LetsEncrypt;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extensions to enable Azure KeyVault connections with LetsEncrypt.
+    /// </summary>
+    public static class AzureKeyVaultExtensions
+    {
+        /// <summary>
+        /// Adds key vault certificate repository for LetsEncrypt.
+        /// </summary>
+        /// <param name="builder">A LetsEncrypt service builder.</param>
+        /// <param name="config">Configuration for KeyVault connections.</param>
+        /// <returns>The original LetsEncrypt service builder.</returns>
+        public static ILetsEncryptServiceBuilder AddKeyVaultCertificateRepository(this ILetsEncryptServiceBuilder builder, Action<AzureKeyVaultCertificateRepositoryOptions>? config = null)
```

Comment:
```suggestion
        public static ILetsEncryptServiceBuilder AddKeyVaultCertificateRepository(this ILetsEncryptServiceBuilder builder)
                    => builder.AddKeyVaultCertificateRepository(_ => {});
                   
        public static ILetsEncryptServiceBuilder AddKeyVaultCertificateRepository(this ILetsEncryptServiceBuilder builder, Action<AzureKeyVaultCertificateRepositoryOptions> config)
```

Nit: can we split this into two overloads? I've found that using optional parameters in the long run actually makes it more difficult to extend and API with new parameters in the future.

Reply:
Sure, no problem